File: VB/WcfService1/Service1.svc.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Xpo.DB
Imports DevExpress.Xpo
Imports DevExpress.Xpo.Metadata
Imports ClassLibrary1

Namespace WcfService1
	Public Class Service1
		Inherits SerializableObjectLayerService
		Public Sub New()
			MyBase.New(New ObjectServiceProxy())
		End Sub
		Shared Sub New()
			Dim connectionString As String = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB")
			Dim dataStore As IDataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema)
			Dim dictionary As XPDictionary = New ReflectionDictionary()
			dictionary.CollectClassInfos(GetType(Customer).Assembly)
			XpoDefault.DataLayer = New ThreadSafeDataLayer(dictionary, dataStore)
			XpoDefault.Session = Nothing
		End Sub
		Public Sub New(ByVal serializableObjectLayer As ISerializableObjectLayer)
			MyBase.New(serializableObjectLayer)
		End Sub
	End Class

	Public Class ObjectServiceProxy
		Inherits SerializableObjectLayerProxyBase
		Protected Overrides Function GetObjectLayer() As SerializableObjectLayer
			Return New SerializableObjectLayer(New UnitOfWork(), True)
		End Function
	End Class
End Namespace

File: CS/WcfService1/Service1.svc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpo.DB;
using DevExpress.Xpo;
using DevExpress.Xpo.Metadata;
using ClassLibrary1;

namespace WcfService1 {

    public class Service1 : SerializableObjectLayerService {
        public Service1()
            : base(new ObjectServiceProxy()) {
        }
        static Service1() {
            string connectionString = MSSqlConnectionProvider.GetConnectionString("localhost", "ServiceDB");
            IDataStore dataStore = XpoDefault.GetConnectionProvider(connectionString, AutoCreateOption.DatabaseAndSchema);
            XPDictionary dictionary = new ReflectionDictionary();
            dictionary.CollectClassInfos(typeof(Customer).Assembly);
            XpoDefault.DataLayer = new ThreadSafeDataLayer(dictionary, dataStore);
            XpoDefault.Session = null;
        }
        public Service1(ISerializableObjectLayer serializableObjectLayer)
            : base(serializableObjectLayer) {
        }
    }

    public class ObjectServiceProxy : SerializableObjectLayerProxyBase {
        protected override SerializableObjectLayer GetObjectLayer() {
            return new SerializableObjectLayer(new UnitOfWork(), true);
        }
    }
}

